python/cucim/tests/performance/clara/test_read_region_memory_usage.py

```diff
@@ -22,23 +23,25 @@
 
 
 def test_read_region_cuda_memleak(testimg_tiff_stripe_4096x4096_256_jpeg):
-    import GPUtil
-
-    gpus = GPUtil.getGPUs()
-
-    if len(gpus) == 0:
+    def get_used_gpu_memory_mib():
+        """Get the used GPU memory in MiB."""
+        status, free, total = cuda.cudart.cudaMemGetInfo()
+        if status != cuda.cudart.cudaError_t.cudaSuccess:
+            raise RuntimeError("Failed to get GPU memory info.")
+        memory_used = (total - free) / (2**20)
+        return memory_used
+
+    status, num_gpus = cuda.cudart.cudaGetDeviceCount()
+    if status != cuda.cudart.cudaError_t.cudaSuccess or num_gpus == 0:
```

Comment:
We can also potentially just use CuPy for this and avoid having to add the cuda-python dependency:

add up top:
```py
import cupy as cp
```

then can use here:
```py
    def get_used_gpu_memory_mib():
        """Get the used GPU memory in MiB."""
        dev = cp.cuda.Device()
        free, total = dev.mem_info
        memory_used = (total - free) / (2**20)
        return memory_used

    num_gpus = cp.cuda.runtime.getDeviceCount()
    if num_gpus == 0:
```

Reply:
Great. I'll implement that and remove the cuda-python dependency.